moveit_ros/visualization/robot_state_rviz_plugin/src/robot_state_display.cpp

```diff
@@ -416,7 +421,10 @@ void RobotStateDisplay::update(float wall_dt, float ros_dt)
   if (load_robot_model_)
   {
     loadRobotModel();
+    // The following call to changedRobotStateTopic() should not change visibility
+    bool visible = robot_->isVisible();
     changedRobotStateTopic();
+    robot_->setVisible(visible);
```

Comment:
Why is that needed? What might change the visibility when changing the topic?

Reply:
`changedRobotStateTopic()` resets the robot to invisible to indicate that we didn't receive a RobotState message on the (new) topic yet. However, here we want to keep the current visibility state.